internal/pkgmanager/ecosystem.go

```diff
@@ -87,15 +98,17 @@ func (p *PkgManager) DownloadArchive(name, version, directory string) (string, e
 		return "", fmt.Errorf("no url found for package %s, version %s", name, version)
 	}
 
-	fileName := p.archiveFilename(name, version, downloadURL)
-
-	archivePath, err := downloadToDirectory(directory, downloadURL, fileName)
+	baseFilename := p.archiveFilename(name, version, downloadURL)
+	if baseFilename == "" {
+		panic("base filename for archive is empty")
```

Comment:
When would this happen? When an invalid URL is passed? Do we need to document that?

Reply:
It shouldn't happen, but it's just a guard that makes debugging easier if one of the archiveFilename functions is incorrectly implemented and returns an empty string. Happy to remove this if you prefer

Reply:
Sure, thanks for explaining. Let's just leave this here then.